tests/test_startup.py

```diff
@@ -69,12 +69,15 @@ def test_on_end(tiny_text_config):
     task_manager.clear_worker_cache.assert_called_once()
 
 
-def test_startup_task_no_train(tiny_text_config_no_train, tiny_text_task_manager):
-    dms = load_dataset_split_managers_from_config(tiny_text_config_no_train)
+def test_startup_task_one_ds(tiny_text_config_one_ds, tiny_text_task_manager):
+    dms = load_dataset_split_managers_from_config(tiny_text_config_one_ds)
     assert DatasetSplitName.eval in dms and DatasetSplitName.train in dms
 
     mods = startup_tasks(dms, tiny_text_task_manager)
-    assert all("train" not in k or "eval" in k for k in mods.keys())
+    ds_name, other_ds_name = get_tiny_text_config_one_ds_name(tiny_text_config_one_ds)
+    assert all(
+        (DatasetSplitName.all or ds_name in k) and other_ds_name not in k for k in mods.keys()
```

Comment:
`DatasetSplitName.all or ds_name in k` will always return `DatasetSplitName.all` because it is truthy, so it is not testing anything. The `in` operator has precedence over the `or`, but even if we added some parentheses, `(DatasetSplitName.all or ds_name) in k` would be equivalent to `DatasetSplitName.all in k` since `DatasetSplitName.all` is truthy, which is not what the correct assertion. We need 2 `in`s:
```suggestion
        (DatasetSplitName.all in k or ds_name in k) and other_ds_name not in k for k in mods.keys()
```

Reply:
Oh great catch!! Thank you, I was struggling with how the precedence would work.